### DISCUSSION TITLE
IPOPT Mex File Compilation

### DISCUSSION THREAD

**Discussion by anilvrao** [coin-or/Ipopt#705] [Q&A] [answered]
I know that MATLAB Mex files have been compiled previously and that Enrico Bertolazzi did an amazing job.  My question, however, is compiling new Mex files for the recent versions of MATLAB including the new version for the M1/M2 Mac processors.  

Does anyone know how to compile Mex files for IPOPT for the recent versions of MATLAB?

**Comment by rdzman** [comment, marked as answer]
I was able to build a native MEX version for Octave on my M2 Mac, but not yet for the latest MATLAB.

FWIW, here is the related [mexIPOPT issue](https://github.com/ebertolazzi/mexIPOPT/issues/24).
